apps/etl-func/src/adapters/pdv-tokenizer/pdv-tokenizer-client.ts

```diff
@@ -0,0 +1,70 @@
+import { TokenizerClient } from "@/domain/interfaces/tokenizer.js";
+import * as assert from "node:assert";
+
+import { piiResourceSchema } from "./pii-resource.js";
+import { problemSchema } from "./problem.js";
+import { tokenResourceSchema } from "./token-resource.js";
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export default class PDVTokenizerClient implements TokenizerClient {
+  #apiKey: string;
+  #baseUrl: string;
+
+  constructor(apiKey: string, baseUrl: string) {
+    assert.ok(apiKey, new Error("Api key is required"));
+    this.#apiKey = apiKey;
+    this.#baseUrl = baseUrl;
+  }
+
+  async #retry<T>(
+    fn: () => Promise<T>,
+    retries = 3,
+    delayMs: 1000,
+    retryCondition: (error: unknown) => boolean = () => true,
+  ): Promise<T> {
+    let attempt = 0;
+    while (attempt < retries) {
+      try {
+        return await fn();
+      } catch (error) {
+        attempt++;
+        if (!retryCondition(error) || attempt >= retries) {
+          throw error;
+        }
+        await delay(delayMs);
+      }
+    }
+    throw new Error(`Error during tokenizer api call after ${retries} retries`);
+  }
+
+  async maskSensitiveInfo(pii: string): Promise<string> {
```

Comment:
```suggestion
  async tokenize(pii: string): Promise<string> {
```

or `createToken` (`POST /tokens`)